src/images/txb.cpp

```diff
@@ -177,18 +159,27 @@ void TXB::readHeader(Common::SeekableReadStream &txb, byte &encoding) {
 
 		_mipMaps.push_back(mipMap.release());
 
-		width      >>= 1;
-		height     >>= 1;
-		mipMapSize >>= 2;
-
-		if ((width < 1) && (height < 1))
-			break;
+		if (width  > 1) width  >>= 1;
+		if (height > 1) height >>= 1;
```

Comment:
This now favours adding a whole lot of 1x1 mipmaps if the file specifies them. Does it really matter?